src/cdk/scrolling/viewport-ruler.ts

```diff
@@ -77,27 +78,20 @@ export class ViewportRuler implements OnDestroy {
     };
   }
 
-  /**
-   * Gets the (top, left) scroll position of the viewport.
-   * @param documentRect
-   */
-  getViewportScrollPosition(documentRect: ClientRect | undefined = this._documentRect) {
-    // Cache the document bounding rect so that we don't recompute it for multiple calls.
-    if (!documentRect) {
-      this._cacheViewportGeometry();
-      documentRect = this._documentRect;
-    }
-
+  /** Gets the (top, left) scroll position of the viewport. */
+  getViewportScrollPosition() {
     // The top-left-corner of the viewport is determined by the scroll position of the document
     // body, normally just (scrollLeft, scrollTop). However, Chrome and Firefox disagree about
     // whether `document.body` or `document.documentElement` is the scrolled element, so reading
     // `scrollTop` and `scrollLeft` is inconsistent. However, using the bounding rect of
     // `document.documentElement` works consistently, where the `top` and `left` values will
     // equal negative the scroll position.
-    const top = -documentRect!.top || document.body.scrollTop || window.scrollY ||
+    const documentRect = document.documentElement.getBoundingClientRect();
```

Comment:
Not for this PR, but thinking about this again, I wonder if there's a better way to get what we're after here than calling `getBoundingClientRect` here.

Reply:
AFAIK the cross-browser way to do it was something like this: `window.pageYOffset || document.documentElement.scrollTop || document.body.scrollTop || 0`.